src/oncall/api/v0/oncall_events.py

```diff
@@ -0,0 +1,169 @@
+
+        HTTP/1.1 200 OK
+        Content-Type: application/json
+
+        {
+            "ldap_user_id":
+                [
+                    {
+                        "start": 1488441600,
+                        "end": 1489132800,
+                        "team": "foo-sre",
+                        "link_id": null,
+                        "schedule_id": null,
+                        "role": "primary",
+                        "user": "foo",
+                        "full_name": "Foo Icecream",
+                        "id": 187795
+                    },
+                    {
+                        "start": 1488441600,
+                        "end": 1489132800,
+                        "team": "foo-sre",
+                        "link_id": "8a8ae77b8c52448db60c8a701e7bffc2",
+                        "schedule_id": 123,
+                        "role": "primary",
+                        "user": "bar",
+                        "full_name": "Bar Apple",
+                        "id": 187795
+                    }
+                ]
+        ]
+
+    :query team: team name
+    :query user: user name
+    :query role: role name
+    :query id: id of the event
+    :query start: start time (unix timestamp) of event
+    :query end: end time (unix timestamp) of event
+    :query start__gt: start time (unix timestamp) greater than
+    :query start__ge: start time (unix timestamp) greater than or equal
+    :query start__lt: start time (unix timestamp) less than
+    :query start__le: start time (unix timestamp) less than or equal
+    :query end__gt: end time (unix timestamp) greater than
+    :query end__ge: end time (unix timestamp) greater than or equal
+    :query end__lt: end time (unix timestamp) less than
+    :query end__le: end time (unix timestamp) less than or equal
+    :query role__eq: role name
+    :query role__contains: role name contains param
+    :query role__startswith: role name starts with param
+    :query role__endswith: role name ends with param
+    :query team__eq: team name
+    :query team__contains: team name contains param
+    :query team__startswith: team name starts with param
+    :query team__endswith: team name ends with param
+    :query team_id: team id
+    :query user__eq: user name
+    :query user__contains: user name contains param
+    :query user__startswith: user name starts with param
+    :query user__endswith: user name ends with param
+
+    :statuscode 200: no error
+    :statuscode 400: bad request
+    """
+    req.params.pop('fields', None)
+    req.params.pop('include_subscribed', None)
+    if any(key not in constraints for key in req.params):
+        raise HTTPBadRequest('Bad constraint param')
+    get_events(req, resp)
+    env = application_environment()
+    config_file_path = os.path.join(env.config_path, 'oncall-api.yaml')
+    with open(config_file_path) as f:
+        config = yaml.safe_load(f)
+    oncall_blacklist = set(config.get('oncall_blacklist', []))
```

Comment:
perhaps rename to oncall_bonus_blacklist and oncall_bonus_whitelist for clarity.